pkgs/unified_analytics/lib/src/initializer.dart

```diff
@@ -86,6 +85,8 @@ class Initializer {
     sessionFile.createSync(recursive: true);
     sessionFile
         .writeAsStringSync('{"session_id": ${now.millisecondsSinceEpoch}}');
+    // TODO: eliasyishak, remove the below once https://github.com/google/file.dart/issues/236 is fixed
+    sessionFile.setLastModifiedSync(clock.now());
```

Comment:
Referring to:
- https://github.com/google/file.dart/issues/236

Reply:
Are you doing this solely for test code? Let's try to avoid doing extra file system operations just for the sake of tests.

Reply:
Yes, a test consistently fails if we don't set the modified timestamp because it thinks there is more than one session since the last modified timestamp gets set to something like `2000-01-01 00:08:00.000`

https://github.com/eliasyishak/tools/blob/remove-late-variables/pkgs/unified_analytics/test/unified_analytics_test.dart#L1089

I don't think deleting a valid test because of a bug would is ideal in this case either.

Reply:
Actually this is not a bug, the dartdocs explicitly say that a fake clock is used for the `MemoryFileSystem.test` constructor. Currently trying to refactor around this